Automattic/newspack-newsletters
# feat: handle post status through `is_public` meta

### All Submissions:






### Changes proposed in this Pull Request:

This PR introduces "controlled statuses". `publish` or `private` statuses changes are now capable of publishing a newsletter. The `is_public` meta will determine which is the final status of a sent newsletter.

This approach introduces the use of `private` status to control newsletter visibility and enables simplification of how we hook into global queries to avoid non-public newsletters from being displayed.

### How to test the changes in this Pull Request:

1. Check out this branch, make sure you have an ESP configure and send a newsletter that will not be public
2. Confirm the newsletter is sent and the post is `private`
3. Edit the sent newsletter, toggle the "Make newsletter page public?" option, update and confirm the post is now "publish"
4. Toggle off the option, save and confirm it's back to private
5. Send the newsletter to the trash and restore
6. Confirm the restored newsletter is set to private
7. Send a new newsletter with "Make newsletter page public?" toggled on and confirm it's sent with status set to "publish"

### Other information:
